calico_containers/integrations/kubernetes/tests/kube_plugin_test.py

```diff
@@ -0,0 +1,414 @@
+
+
+    def test_get_pod_ports(self):
+        # Initialize pod dictionary
+        pod = {'spec':{'containers':[{'ports':[1,2,3]},{'ports':[4,5]}]}}
+        ports = []
+
+        # Append port numbers to a list for comparison
+        for container in pod['spec']['containers']:
+            ports.extend(container['ports'])
+
+        # Call method under test
+        return_val = self.plugin._get_pod_ports(pod)
+
+        # Assert
+        self.assertEqual(return_val,ports)
+
+    def test_get_pod_ports_fail(self):
+        # Initialize pod dictionary
+        pod = {'spec':{'containers':[{'':[1,2,3]},{'':[4,5]}]}}
+        ports = []
+
+        # Call method under test
+        return_val = self.plugin._get_pod_ports(pod)
+
+        # Assert
+        self.assertListEqual(return_val,ports)
+
+    def test_get_pod_config(self):
+        with patch.object(self.plugin, '_get_api_path',
+                          autospec=True) as m_api_path:
+            # Set up mock object
+            pod1 = {'metadata':{'name':'pod-1'}}
+            pod2 = {'metadata':{'name':'pod-2'}}
+            pod3 = {'metadata':{'name':'pod-3'}}
+            pods = [pod1,pod2,pod3]
+            m_api_path.return_value = pods
+
+            # Set up class member
+            self.plugin.pod_name = 'pod_2'
+
+            # Call method under test
+            return_val = self.plugin._get_pod_config()
+
+            # Assert
+            self.assertEqual(return_val,pod2)
+
+    def test_get_pod_config_fail(self):
+        with patch.object(self.plugin, '_get_api_path',
+                          autospec=True) as m_api_path:
+            # Set up mock object and class members
+            pod1 = {'metadata':{'name':'pod-1'}}
+            pod2 = {'metadata':{'name':'pod-2'}}
+            pod3 = {'metadata':{'name':'pod-3'}}
+            pods = [pod1,pod2,pod3]
+            m_api_path.return_value = pods
+
+            # Set up class member
+            self.plugin.pod_name = 'pod_4'
+
+            # Call method under test expecting exception
+            with self.assertRaises(KeyError):
+                self.plugin._get_pod_config()
+
+
+    def test_get_api_path(self):
+        with patch.object(self.plugin, '_get_api_token',
+                          autospec=True) as m_api_token, \
+                patch('calico_containers.integrations.kubernetes.'
+                      'calico_kubernetes.requests.Session',
+                      autospec=True) as m_session, \
+                patch.object(json, 'loads',autospec=True) as m_json_load:
+            # Set up mock objects
+            m_api_token.return_value = 'Token'
+            m_session_return = Mock()
+            m_session_return.headers = Mock()
+            m_get_return = Mock()
+            m_get_return.text = 'response_body'
+            m_session_return.get.return_value = m_get_return
+            m_session.return_value = m_session_return
+
+            # Initialize args
+            path = 'path'
```

Comment:
Make the path something that looks more like a path we'd expect.


Perhaps something like `/path/to/api/object`